packages/lisk-p2p/src/peer_pool.ts

```diff
@@ -228,17 +236,32 @@ export class PeerPool extends EventEmitter {
 			// TODO ASAP: We need a new type to account for complete P2PPeerInfo which has all possible fields (e.g. P2PDiscoveredPeerInfo) that way we don't need to have all these checks below.
 			peers: this._pickRandomPeers(MAX_PEER_LIST_BATCH_SIZE).map(
 				(peer: Peer) => {
+					const peerDetailedInfo = peer.detailedPeerInfo;
 					const peerInfo = peer.peerInfo;
 
+					if (peerDetailedInfo) {
+						return {
+							broadhash: peerDetailedInfo.options
+								? (peerDetailedInfo.options.broadhash as string)
+								: '',
+							height: peerDetailedInfo.height,
+							ip: peerDetailedInfo.ipAddress,
+							nonce: peerDetailedInfo.options
+								? (peerDetailedInfo.options.nonce as string)
+								: '',
+							os: peerDetailedInfo.os ? peerDetailedInfo.os : '',
+							version: peerDetailedInfo.version ? peerDetailedInfo.version : '',
+							wsPort: String(peerInfo.wsPort),
+						};
+					}
+
 					return {
-						broadhash: peerInfo.options
-							? (peerInfo.options.broadhash as string)
-							: '',
+						broadhash: '',
```

Comment:
As mentioned before, no need to return unless we have complete discovered peer info; ideally, this case should never happen.

Reply:
I had to return it because of the above comment, I had to keep a default return statement

Reply:
In that case, maybe it can return null and then we can add a `filter` function to remove all the null references from the mapped list? Or we can use filter and map before so that we are sure that this function only handles `DiscoveredPeerInfo` objects.

We should never send back a peer which has not been discovered yet so the code should not make any effort to support this.